python/ray/tests/test_failure.py

```diff
@@ -628,6 +629,39 @@ def check_actor_restart():
     ray.get(obj2)
 
 
+def test_final_user_exception(ray_start_regular, propagate_logs, caplog):
+    class MyFinalException(Exception):
+        def __init_subclass__(cls, /, *args, **kwargs):
+            raise TypeError("Cannot subclass special typing classes")
+
+    # This should error.
+    with pytest.raises(MyFinalException):
+        raise MyFinalException("MyFinalException from driver")
+
+    @ray.remote
+    def func():
+        # This should also error. Problem is, the user exception is final so we can't
+        # subclass it (raises exception if so). This means Ray cannot raise an exception
+        # that can be caught as both `RayTaskError` and the user exception. So we
+        # issue a warning and just raise it as `RayTaskError`. User needs to use
+        # `e.cause` to get the user exception.
+        raise MyFinalException("MyFinalException from task")
+
+    with caplog.at_level(logging.WARNING, logger="ray.exceptions"):
+        with pytest.raises(ray.exceptions.RayTaskError) as exc_info:
+            ray.get(func.remote())
+
+    assert (
+        "This exception will be raised as RayTaskError only. User can use "
```

Comment:
```suggestion
        "This exception is raised as RayTaskError only. The user can use "
```

Reply:
Not sure if "The user" is correct. If "user" is the reader, then use "you", or just simply say "Use ... to access the user exception."

Reply:
It's `You` the reader. Updated.